lib/vdc/BOVCollection.cpp

```diff
@@ -0,0 +1,500 @@
+            value[i] = lineValue;
+        }
+
+        if (lineStream.bad()) {
+            // value.clear();
+            std::string message = "Invalid value for " + token + " in BOV header file.";
+            SetErrMsg(message.c_str());
+            return ERROR;
+        }
+
+        if (verbose) {
+            std::cout << std::setw(20) << token << " ";
+            for (int i = 0; i < value.size(); i++) std::cout << value[i] << " ";
+            std::cout << std::endl;
+        }
+        return FOUND;
+    }
+    return NOT_FOUND;
+}
+
+void BOVCollection::_findTokenValue(std::string &line) const
+{
+    std::string delimiter = ": ";
+
+    size_t      pos = 0;
+    std::string token;
+    while ((pos = line.find(delimiter)) != std::string::npos) {
+        token = line.substr(0, pos);
+        line.erase(0, pos + delimiter.length());
+    }
+}
+
+int BOVCollection::_sizeOfFormat(DC::XType type) const
+{
+    switch (type) {
+    case DC::XType::INT32: return 4;
+    case DC::XType::FLOAT: return 4;
+    case DC::XType::DOUBLE: return 8;
+    default: return -1;
+    }
+}
+
+void BOVCollection::_swapBytes(void *vptr, size_t size, size_t n) const
+{
+    unsigned char *ucptr = (unsigned char *)vptr;
+    unsigned char  uc;
+    size_t         i, j;
+
+    for (j = 0; j < n; j++) {
+        for (i = 0; i < size / 2; i++) {
+            uc = ucptr[i];
+            ucptr[i] = ucptr[size - i - 1];
+            ucptr[size - i - 1] = uc;
+        }
+        ucptr += size;
+    }
+}
+
+template<class T> int BOVCollection::ReadRegion(std::string varname, size_t ts, const std::vector<size_t> &min, const std::vector<size_t> &max, T region)
+{
+    float       time = _times[ts];
+    std::string dataFile = _dataFileMap[varname][time];
+
+    FILE *fp = fopen(dataFile.c_str(), "rb");
+    if (!fp) {
+        SetErrMsg("Invalid file: %d", fp);
+        return -1;
+    }
+
+    int formatSize = _sizeOfFormat(_dataFormat);
+    if (formatSize < 0) {
+        SetErrMsg("Unspecified data format");
+        fclose(fp);
+        return (-1);
+    }
+
+    size_t numValues = _gridSize[0] * _gridSize[1] * _gridSize[2];
+
+    int  n = 1;
+    bool systemLittleEndian = *(char *)&n == 1 ? true : false;
+    bool dataLittleEndian = _dataEndian == "LITTLE" ? true : false;
+    bool needSwap = systemLittleEndian != dataLittleEndian ? true : false;
+
+    // Read a "pencil" of data along the X axis, one row at a time
+    size_t count = max[0] - min[0] + 1;
+    // Re-use the same buffer for all the pencils.
+    std::vector<unsigned char> vReadBuffer(count * formatSize);
+    unsigned char *            readBuffer = vReadBuffer.data();
+    for (int k = min[2]; k <= max[2]; k++) {
+        int zOffset = _gridSize[0] * _gridSize[1] * k;
+        for (int j = min[1]; j <= max[1]; j++) {
+            int xOffset = min[0];
+            int yOffset = _gridSize[0] * j;
+            int offset = formatSize * (xOffset + yOffset + zOffset);
```

Comment:
Can't you simply add _byteOffset to offset and have only a single seek with SEEK_SET? The code as is will work, but it's a little confusing.

Reply:
Yep.  Fixed.